### PR TITLE
[zeromq | thrift] Replace deprecated cmake variables

### LINKED ISSUE #31099: [thrift] CMake variable misspelled
~~~
$ git grep CMAKE_FIND_PACKAGE_REQUIRE ports/thrift/
ports/thrift/portfile.cmake:        -DCMAKE_FIND_PACKAGE_REQUIRE_ZLIB=TRUE
ports/thrift/portfile.cmake:        -DCMAKE_FIND_PACKAGE_REQUIRE_Libevent=TRUE
ports/thrift/portfile.cmake:        -DCMAKE_FIND_PACKAGE_REQUIRE_OpenSSL=TRUE
ports/thrift/portfile.cmake:        CMAKE_FIND_PACKAGE_REQUIRE_Libevent
ports/thrift/portfile.cmake:        CMAKE_FIND_PACKAGE_REQUIRE_OpenSSL
ports/thrift/portfile.cmake:        CMAKE_FIND_PACKAGE_REQUIRE_ZLIB
~~~
Proper spelling: `CMAKE_REQUIRE_FIND_PACKAKE_<pkg>`

### LINKED ISSUE #31100: [zeromq] CMake variable misspelled
~~~
$ git grep CMAKE_FIND_PACKAGE_REQUIRE ports/zeromq/
ports/zeromq/portfile.cmake:        -DCMAKE_FIND_PACKAGE_REQUIRE_GnuTLS=ON
~~~
Proper spelling: `CMAKE_REQUIRE_FIND_PACKAKE_<pkg>`

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fixes #31100, Fix #31099.

Replace deprecated cmake variable from `CMAKE_FIND_PACKAGE_REQUIRE_<pkg>` to `CMAKE_REQUIRE_FIND_PACKAGE_<pkg>`.

### CODE REVIEW

**Comment by huangqinjin** [comment]
emmm... should be `CMAKE_REQUIRE_FIND_PACKAGE_<pkg>` ?

**Review by LilyWangLL — APPROVED** [review]

**Comment by BillyONeal** [comment]
Oops! Thanks!

File: ports/zeromq/portfile.cmake [hunk 1]
```diff
@@ -34,8 +34,7 @@ vcpkg_cmake_configure(
         -DWITH_PERF_TOOL=OFF
         -DWITH_DOCS=OFF
         -DWITH_NSS=OFF
-        -DCMAKE_FIND_PACKAGE_REQUIRE_GnuTLS=ON
-        -DWITH_LIBBSD=OFF
```
Comment (dg0yt) [inline ports/zeromq/portfile.cmake:38]: Didn't we add this as a fix only a few days ago?

Comment (BillyONeal) [reply ports/zeromq/portfile.cmake:38]: Oops, thanks!

Comment (BillyONeal) [reply ports/zeromq/portfile.cmake:38]: https://github.com/microsoft/vcpkg/pull/31151

**Pull request merged**